### PR TITLE
Refactor controllers

### PR DESCRIPTION
**Pull request by Ru-T**
## Assignment
- To refactor and fix common mistakes in routes and controllers (views are next)
## What Changed?
- Edited routes file to use resources
- Moved validations out of items controller and into items model, with tests and factory
## Why?
- Restful routes
- Fat models, skinny controllers

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 class Item < ActiveRecord::Base
   belongs_to :category
 
+  validates :name, presence: true
```
Comment (joeyjoejoejr) [inline app/models/item.rb]: I don't know whether it's in vogue now, but there is another way to do this kind of validation

``` ruby
validates_presence_of :name, :price, :quantity, :shelf_life_days
```

So if you have the same validation on multiple fields you can do it this way.  The `validates` method is better if you have multiple validations and configurations on one attribute.

File: config/routes.rb [hunk 2]
```diff
@@ -1,9 +1,6 @@
 Rails.application.routes.draw do
   root 'items#home'
 
-  get '/home', to: 'items#home'
-  post '/items', to: 'items#create'
-  patch '/items', to: 'items#update'
-
-  post '/categories', to: 'categories#create'
+  resources :items, only: [:create]
```
Comment (joeyjoejoejr) [inline config/routes.rb]: You also need an update and home route here, I think.

**Comment by joeyjoejoejr** [comment]
Looks great.  I think you do have one thing that is an actual error though.  You will no longer have the routes `items#update` or `items#home`

File: app/models/item.rb (hunk 1 above)
Comment (Ru-T) [inline app/models/item.rb]: Nice. I usually do one line per validation to make it more readable, but this is actually super readable. Changed it!

File: config/routes.rb (hunk 2 above)
Comment (Ru-T) [inline config/routes.rb]: Thanks, good catch. There is actually no update action in the controller so I thought this was unintentional - but I added back.

Comment (joeyjoejoejr) [inline config/routes.rb]: Oh... um.  @BrandonMathis that's a problem.

**Pull request merged**